### DISCUSSION TITLE
Use of GPS for Time Synchronization

### DISCUSSION THREAD

**Discussion by Ken-Kauffmann** [arnaud-j4k08/PicoClockGreenEasy#7] [Ideas]
I am excited that you are exploring the use of GPS for time and date synchronization.  I am creating this discussion so that you can remember to let me know how you are progressing.  I don't want to miss out. :-)

It is important for the end-user to be able to set Time and Date manually, without any need to compile the program, or have Wifi available.  I am glad that you started with that, and I think that it remains a key feature going forward. 

The ability to synchronize with NTP over the Wifi connection is really nice.  I like that you default to having that update done once at initialization, so that the end-user still has the ability to play with the time and date as they wish.  I did make the changes in my version to attempt to re-sync each day, at a user-selected time.  ( I am doing it at 21 minutes after midnight.)  Like you, I have concerns that the Wifi session will not always be active at that time, but I still a good option. 

I think that GPS will be even better.  As you described, the GPS units and antennas fit nicely into the PicoClockGreen case, and we have easy access to the Pico pins. In my experience, once the GPS locks on, it stays locked on, and will provide an excellent time source. 

Please let me know if I can help in any way.  Thanks, Ken

**Comment by arnaud-j4k08** [comment]
I have pushed my last commits to github, so that you can see the current state of my progress. It also brings other changes that we have discussed separately. I will comment in the respective threads.

I have added a "clock sync" submenu that current allows:
- selecting a sync source among RTC, NTP and GPS.
- triggering a synchronization manually

Daily resync is not implemented yet, but GPS synchronization is working. You may want to try it, as it is a great coincidence that you already have the required hardware. As documented in README.md, the code expects this wiring:
|GPS module|Pico    |
|----------|--------|
|VCC       |3V3(OUT)|
|RX        |GP0     |
|TX        |GP1     |
|GND       |GND     |

I did not dare putting the antenna in the PicoClockGreen case like the GPS module, as I feared the reception would be worse. But I will try at the next occasion. Maybe it does not make any difference, and the integration will be even better.

**Reply by arnaud-j4k08** [reply]
BTW, I have put the antenna into the case in the meantime. Maybe it takes a little bit more time to acquire satellites, but overall it makes the integration almost perfect. After acquisition, a red led keeps blinking through the case. This slightly annoys me, so I am considering disabling it. It looks like the protocol of the GPS modules allows this.

**Comment by Ken-Kauffmann** [comment]
Fantastic implementation.  Thank you.  Ken
